homeassistant/components/sensor/statistics.py

```diff
@@ -71,11 +77,18 @@ def __init__(self, hass, entity_id, name, sampling_size):
         else:
             self._name = '{} {}'.format(name, ATTR_COUNT)
         self._sampling_size = sampling_size
+        self._max_age = max_age
         self._unit_of_measurement = None
         if self._sampling_size == 0:
             self.states = deque()
         else:
             self.states = deque(maxlen=self._sampling_size)
+        if self._max_age is not None:
+            if self._sampling_size == 0:
+                self.ages = deque()
```

Comment:
I agree with @pvizeli and will make that change - I'm currently just a bit short on time. :)